### ISSUE TITLE
[9.0.0] Filters don't work while HyperFormula is enabled as new formulas engine

### ISSUE THREAD

**Issue by aninde** [handsontable/handsontable#8049]
### Description
Filtering by conditions and by value doesn't work while HyperFormula is enabled as a formulas engine.

#### Case 1
Filtering by conditions doesn't work. It seems to filter out all rows besides the first one. 
![filtr-by-cond](https://user-images.githubusercontent.com/30186818/118527181-531b7e00-b741-11eb-836b-64e876234bf0.gif)

#### Case 2
Filtering by value doesn't work. In all my cases always the last row was filtered out.
![filt1](https://user-images.githubusercontent.com/30186818/118527191-557dd800-b741-11eb-9774-11b7d4561610.gif)

### Demo

https://codesandbox.io/s/filtering-withhyperformula-forked-tl5c7 v9.0.0 from [PR ](https://github.com/handsontable/handsontable/pull/7648)+ HF v.0.6.0
 https://codesandbox.io/s/filtering-forked-ud0wh working old engine

### Your environment
* Handsontable version: v9.0.0 from [PR ](https://github.com/handsontable/handsontable/pull/7648)+ HF v.0.6.0
* Browser Name and version: Chrome 91
* Operating System: macOS

**Comment by aninde** [comment]
![image](https://user-images.githubusercontent.com/30186818/118610814-75061680-b7bc-11eb-8bbc-ad2ad8f077bc.png)

Filtering doesn't need to work with formulas as was written in the docs.
However documentation was no up to date, cause filters used to work with old formulas engine.

**Comment by nikersify** [comment]
Should be fixed after #8067

**Comment by aninde** [comment]
This bug is fixed on current v9.0.0 https://jsfiddle.net/aninde/ktpwcbas/
More advanced operations like filtering moved rows can cause truncation of column headers. 
Those cases will be reported in a separate issue.
